subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -275,9 +275,11 @@ int ll_tx_mem_enqueue(uint16_t handle, void *tx)
 #endif /* CONFIG_BT_CTLR_FORCE_MD_AUTO */
 	}
 
-	if (IS_ENABLED(CONFIG_BT_PERIPHERAL) && conn->lll.role) {
+#if defined(CONFIG_BT_PERIPHERAL)
```

Comment:
Wouldn't moving `peripheral_latency_cancel` to not be under `#ifdef CONFIG_BT_PERIPHERAL` have solved this?

Atleast that is the approach used in the host, to avoid too much ifdeffed code.